src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/BinaryFormatterUnsafeDeserializeAnalyzer.cs

```diff
@@ -39,71 +40,81 @@ public override void Initialize(AnalysisContext context)
             context.EnableConcurrentExecution();
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
 
-            context.RegisterSyntaxNodeAction(AnalyzeMemberAccessExpressions, SyntaxKind.SimpleMemberAccessExpression);
+            context.RegisterSimpleMemberAccessExpression(AnalyzeMemberAccessExpressions);
         }
 
         private void AnalyzeMemberAccessExpressions(SyntaxNodeAnalysisContext context)
         {
-            var memberAccessExpression = (MemberAccessExpressionSyntax)context.Node;
-
-            // If the accessed member isn't named "UnsafeDeserialize" bail out
-            if (!TargetMember.Equals(memberAccessExpression.Name.ToString(), StringComparison.Ordinal))
+            if (!context.Node.IsMemberAccessExpression())
             {
                 return;
             }
 
-            // Continue, only if the call is to a method called UnsafeDeserialize
-            if (!(memberAccessExpression.Parent is InvocationExpressionSyntax))
+            var expression = new GeneralMemberAccessExpression(context.Node);
```

Comment:
Maybe pass this in if it is a valid expression so this method will only be hit if it's available